### DISCUSSION TITLE
Utilitary function location

### DISCUSSION THREAD

**Discussion by lucasroges** [moby/moby#45093] [Q&A]
I want to implement an utilitary function which will be called by multiple endpoints of different API files.

I'm still a bit confused with the repository structure, so where is the best location to implement such a function ?

**Comment by thaJeztah** [comment]
The API server endpoints can be found in (subdirectories of); https://github.com/moby/moby/tree/master/api/server/router

Can you elaborate a bit on "utilitary function"? What's the feature you had in mind?

**Reply by lucasroges** [reply]
I aim to create a validation function for the `since` and `until` flags from `docker container logs` (and also to be used by any other commands with such flags).

While looking at the code, I noticed that the `docker events` command already has such validation, but all related code is within the endpoint's file.

I want to create a single function separated from these endpoints, but I need clarification about where would be the best location. 

Although assigning contradictory values for these flags is harmless, I encountered a situation in which the log file was very large and it took a while to read it entirely. But in the end, I was waiting for no input since I switched the values from these flags, and thus they were contradictory (e.g., `docker container logs --since "20m" --until "30m" {container}`)
